## Verified locally now — and it turned up something bigger than the 18

The PR body said "not verified locally: mypy is not installed on this host." It is now. Both `pyproject.toml` fixes confirm: the `python_version` rejection notice and the unused-`alembic.*`-section note are both gone from the output.

But running it locally exposed a real defect in the job itself.

### The type check runs blind in CI

`quality-check` installs only:

```
pip install 'black==23.12.1' 'isort==5.13.2' flake8 mypy pre-commit
```

It never installs the project or its dependencies. So mypy has no fastapi, pydantic or sqlalchemy type information, and `--ignore-missing-imports` silently swallows that.

Measured, controlled comparison — same commit, same flags, only the environment differs:

| Environment | Result |
|---|---|
| mypy alone, no project deps (**what CI does**) | **18 errors in 4 files** |
| mypy + `pip install -e '.[dev]'` | **16 errors in 5 files** |

The first reproduces CI exactly. The second reproduces the "16" figure quoted in the last session handoff — so both numbers were always real, they just came from different environments. That discrepancy was previously unexplained.

### What blindness costs

**6 errors CI reports that are artifacts**, not defects — the `api.py` `untyped-decorator` cascade at lines 21, 46, 86, 111, 162, 200. With fastapi installed, those decorators are typed and the errors vanish.

**5 real errors CI structurally cannot see:**

```
src/tick_task/config.py:59: error: Missing named argument "database_url" for "Settings"  [call-arg]
src/tick_task/config.py:59: error: Missing named argument "data_dir" for "Settings"  [call-arg]
src/tick_task/config.py:59: error: Missing named argument "lan_mode" for "Settings"  [call-arg]
src/tick_task/config.py:59: error: Missing named argument "lan_token" for "Settings"  [call-arg]
src/tick_task/database.py:18: error: No overload variant of "sessionmaker" matches argument types "AsyncEngine", "type[AsyncSession]", "bool"  [call-overload]
```

The `database.py:18` one is worth a look on its own merits — an async sessionmaker that does not match any documented overload is the kind of thing that works until it doesn't.

### What I did not change here

I have deliberately **not** added the project install to `quality-check` in this PR. It changes the error set, and this PR's job is to unbreak `main` without smuggling in a behavior change. Recorded in https://github.com/Twin-Cities-Open-Systems/tick-task/issues/44 instead — the flip condition there should include "mypy runs with dependencies installed", because clearing 18 blind errors while 5 real ones stay invisible would satisfy the letter of the gate and none of its purpose.

--- a/src/tick_task/api.py
+++ b/src/tick_task/api.py
@@ -8,7 +8,6 @@ from fastapi import APIRouter, Depends, HTTPException, Query, status
 from sqlalchemy import or_, select
 from sqlalchemy.ext.asyncio import AsyncSession
 
-from tick_task.config import settings
 from tick_task.database import get_db
 from tick_task.models import Task
 from tick_task.schemas import ErrorResponse, HealthResponse
@@ -201,7 +200,10 @@ async def delete_task(
     "/tasks",
     response_model=TaskList,
     summary="List tasks",
-    description="Retrieve a list of tasks with optional filtering, sorting, and pagination",
+    description=(
+        "Retrieve a list of tasks with optional filtering, sorting, "
+        "and pagination"
+    ),
 )
 async def list_tasks(
     # Filtering parameters

--- a/src/tick_task/api.py
+++ b/src/tick_task/api.py
@@ -201,8 +201,7 @@ async def delete_task(
     response_model=TaskList,
     summary="List tasks",
     description=(
-        "Retrieve a list of tasks with optional filtering, sorting, "
-        "and pagination"
+        "Retrieve a list of tasks with optional filtering, sorting, and pagination"
     ),
 )
 async def list_tasks(

--- a/src/tick_task/config.py
+++ b/src/tick_task/config.py
@@ -1,6 +1,5 @@
 """Application configuration."""
 
-import os
 from pathlib import Path
 from typing import Optional
 

--- a/src/tick_task/main.py
+++ b/src/tick_task/main.py
@@ -8,7 +8,6 @@ from fastapi.responses import RedirectResponse
 from tick_task.api import router as api_router
 from tick_task.auth import require_lan_token
 from tick_task.config import settings
-from tick_task.database import create_tables
 
 
 def create_application() -> FastAPI:

--- a/src/tick_task/models.py
+++ b/src/tick_task/models.py
@@ -4,7 +4,7 @@ from datetime import datetime
 from typing import Optional
 from uuid import uuid4
 
-from sqlalchemy import Boolean, Column, DateTime, Enum, Integer, String, Text, func
+from sqlalchemy import DateTime, Enum, String, Text
 from sqlalchemy.dialects.sqlite import JSON
 from sqlalchemy.ext.asyncio import AsyncAttrs
 from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column

--- a/src/tick_task/schemas.py
+++ b/src/tick_task/schemas.py
@@ -2,7 +2,6 @@
 
 from datetime import datetime
 from typing import Optional
-from uuid import UUID
 
 from pydantic import BaseModel, ConfigDict, Field, field_validator
 

